src/overlays/actors/ovl_En_Ginko_Man/z_en_ginko_man.c

```diff
@@ -190,108 +182,83 @@ void EnGinkoMan_DepositDialogue(EnGinkoMan* this, GlobalContext* globalCtx) { //
                 this->curTextId = 0x45A; // "All right, little guy, now I've got a total of [rupees] from you!"
             }
             break;
-        case 10: // "Is that so? Think about it, little guy!"
-        case 13: // "Heyyy! You don't have that much!
+        case 0x456: // "Is that so? Think about it, little guy!"
+        case 0x459: // "Heyyy! You don't have that much!
             func_801518B0(globalCtx, 0x44E, &this->actor);
             this->curTextId = 0x44E; //" ...So, what'll it be?  Deposit Rupees Don't deposit Rupees"
-            return;
-
-        case 14: // "All right, little guy, now I've got a total of [rupees] from you!"
-
-            /***** WARNING: this case is non-equivelent ****/
-
-            tempEDC = gSaveContext.roomInf[127][0] & 0xFFFF;
-            if ((tempEDC >= 200) && (this->previousBankValue < 200) &&
-                ((gSaveContext.weekEventReg[0x3B] & 0x40) == 0)) {
-                // block_20:
-                gSaveContext.weekEventReg[0x3B] = gSaveContext.weekEventReg[0x3B] | 0x40;
+            break;
+        case 0x45A: // "All right, little guy, now I've got a total of [rupees] from you!"
+            if (((gSaveContext.roomInf[127][0] & 0xFFFF) >= 200) && (this->previousBankValue < 200) &&
+                !(gSaveContext.weekEventReg[0x3B] & 0x40)) { // bright pink arrow
+                gSaveContext.weekEventReg[0x3B] |= 0x40;
                 func_801518B0(globalCtx, 0x45B, &this->actor);
                 this->curTextId = 0x45B; // "What's this? You've already saved up 200 Rupees!?!
-                return;
-            } // bright pink arrow
-
-            // "branch likely" when shouldn't be
-            if ((tempEDC >= 1000) && ((this->previousBankValue) < 1000) &&
-                (gSaveContext.weekEventReg[0x3B] & 0x80) == 0) {
-                // block_24:
+            } else if (((gSaveContext.roomInf[127][0] & 0xFFFF) >= 1000) && ((this->previousBankValue) < 1000) &&
+                !(gSaveContext.weekEventReg[0x3B] & 0x80)) { // bright blue arrow
                 gSaveContext.weekEventReg[0x3B] |= 0x80;
                 func_801518B0(globalCtx, 0x45C, &this->actor);
                 this->curTextId = 0x45C; // "What's this? You've already saved up 1000 Rupees!?!
-                return;
-            } // bright blue arrow
-
-            if ((this->previousBankValue) >= 5000) { // added back in because missing
-                if ((((s16)tempEDC) < 5000) && ((gSaveContext.weekEventReg[0x3B] & 1) == 0)) {
-                    gSaveContext.weekEventReg[0x3B] |= 1;
+            } else if ((gSaveContext.roomInf[127][0] & 0xFFFF) >= 5000) {
+                if ((this->previousBankValue < 5000) && !(gSaveContext.weekEventReg[0x3C] & 1)) {
+                    gSaveContext.weekEventReg[0x3C] |= 1; // olive arrow
```

Comment:
I still struggle to follow asm when debugging sometimes, these arrow comments were me keeping track of the branches in diff.py